### PR TITLE
feat(billing): require $100 top-up minimum for trial users

### PR DESCRIPTION
**Pull request by baktun14**
## Why

Trial accounts get $100 of free credits up front. With the existing $20 Stripe top-up floor, a trial user could pay $20, convert to a paying account, and keep the $100 trial credit forever — a net ~$80 loss per such account. Raising the first paid top-up to $100 (matching the trial grant) makes the worst case break-even.

## What

While the user is still on the free trial (`userWallet.isTrialing === true`), the Stripe top-up minimum is **$100**. Once they convert to a paying account, the existing **$20** minimum applies. The transition flips automatically inside the Stripe webhook (`BalancesService.refreshUserWalletLimits({ endTrial: true })`), so gating on `isTrialing` captures "first paid top-up" without needing a separate flag.

**Backend**
- New `MANAGED_WALLET_TRIAL_MIN_TOP_UP_AMOUNT` env var (default `100`).
- `TrialValidationService.getTopUpMinAmountUsd(wallet)` — single source of truth for the minimum (trial → env, otherwise $20).
- `TrialValidationService.validateTopUpAmount(wallet, amount)` — throws 402 with a dynamic message; called from `StripeController.confirmPayment` **before** any Stripe call so a $50 trial top-up never reaches Stripe.
- Wallet response now carries `topUpMinAmountUsd` so the frontend always shows the effective minimum without a duplicate env var.

**Frontend**
- `PaymentPopup` rebuilds its zod schema from `useWallet().topUpMinAmountUsd` and surfaces the active minimum in the input description ("Minimum amount is $100").
- `WalletProvider` exposes `topUpMinAmountUsd` (defaults to `20` when no managed wallet).

**Cleanup**
- Removed the dead `validateTrialLimit` + `TRIAL_DEPLOYMENT_LIMIT` path from `TrialValidationService` — no longer called anywhere — plus the now-orphaned `DeploymentReaderService` constructor dep and related spec scaffolding.

**Tests**
- New `validateTopUpAmount` + `getTopUpMinAmountUsd` specs.
- New `StripeController.confirmPayment` specs covering: trial-min gate is called, rejection short-circuits Stripe, undefined-wallet path.
- `PaymentPopup` spec: dynamic minimum is rendered from the wallet context.
- `WalletController` spec updated to assert the new `topUpMinAmountUsd` field on `getWallets` and the 3DS path.
- All 64 touched API tests + 40 deploy-web tests pass; lint clean.

## Summary by CodeRabbit

* **New Features**
  * Payment top-up enforces and displays a configurable minimum: trial wallets use a trial-specific minimum (configurable, defaults to $100), standard wallets use a $20 minimum.

* **Refactor**
  * Trial validation reorganized to focus on top-up minimum logic and removed legacy trial-deployment limit checks.

* **Tests**
  * Expanded tests to cover top-up minimum behavior and updated wallet/payment UI tests to reflect the new minimum field.

[Review Change Stack](https://app.coderabbit.ai/change-stack/akash-network/console/pull/3211?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/api/src/billing/config/env.config.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export const envSchema = z.object({
     .string()
     .default(AUDITOR)
     .transform(val => (val ? val.split(",").map(addr => addr.trim()) : [])),
+  MANAGED_WALLET_TRIAL_MIN_TOP_UP_AMOUNT: z.number({ coerce: true }).default(100),
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/api/src/billing/config/env.config.ts]: Fixed in 06fbec3 — added `.min(20)` to the schema so empty/zero/negative env values fall below the standard non-trial floor and reject at boot.

File: apps/api/src/billing/services/trial-validation/trial-validation.service.ts [hunk 2]
```diff
@@ -83,6 +70,16 @@ export class TrialValidationService {
...
 
+  getTopUpMinAmountUsd(userWallet: Pick<UserWalletOutput, "isTrialing">): number {
+    return userWallet.isTrialing ? this.config.get("MANAGED_WALLET_TRIAL_MIN_TOP_UP_AMOUNT") : STANDARD_TOP_UP_MIN_AMOUNT_USD;
+  }
+
+  validateTopUpAmount(userWallet: Pick<UserWalletOutput, "isTrialing"> | undefined, amountUsd: number) {
+    if (!userWallet?.isTrialing) return;
+    const min = this.getTopUpMinAmountUsd(userWallet);
+    assert(amountUsd >= min, 402, `First top-up must be at least $${min} while on the free trial.`);
+  }
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/api/src/billing/services/trial-validation/trial-validation.service.ts:85]: Fixed in 06fbec3 — `validateTopUpAmount` now enforces the resolved minimum for every wallet (not just trialing). The amount schema's `>= 20` rejection already caught this in practice, but the validator is now self-consistent with `getTopUpMinAmountUsd`.

**Review by stalniy — APPROVED** [review]
